rest/client.py

```diff
@@ -44,17 +36,43 @@ def order_book(self, currency_pair: str, group: int = 1):
         if group not in [0, 1, 2]:
             raise ValueError('Group parameter should be 0, 1 or 2.')
 
-        rsp = requests.get(url=self.base_endpoint + '/order_book/' + currency_pair, params={'group': group})
-        if rsp.status_code == 404:
-            raise ValueError('Order book for requested currency pair not found.')
+        params = {'group': group}
+        return self._make_request('GET', '/order_book/' + currency_pair, params)
 
-        return rsp.json()
+    def transactions(self, currency_pair: str, period: str = "hour"):
+        """
+        Returns descending list of transactions for specified currency pair. Supports time interval from which we want the transactions
+        to be returned.
+
+        :param currency_pair:  defines which currency pair to get transactions for.
+        :param period: defines the time interval.
+        minute - returns transactions for last minute.
+        hour - returns transactions for the hour.
+        day - returns transactions for the day.
+        """
+        params = {'time': period}
+        return self._make_request('GET', '/transactions/' + currency_pair, params)
 
     def conversion_rate(self):
         """
         Retrieves current BUY/SELL conversion rate for EUR/USD.
         """
-        rsp = requests.get(url=self.base_endpoint + '/eur_usd/')
-        
+        return self._make_request('GET', '/eur_usd/')
+
+    def _make_request(self, method: str, endpoint: str, params: typing.Dict = None, body: typing.Dict = None):
+        """
+        Private function for all different HTTP request methods using request library.
+
+        :param method: indicates the HTTP request method (GET, POST,...)
+        :param endpoint: specific endpoint for for api endpoints.
+        :param data: additional parameters
+        :return: json value of response.
+        """
+        if method == 'GET':
```

Comment:
Sorry, got some more ideas. What If we moved everything that happens in "Happy path" inside try block?
Now you only make requests and raise if something went wrong, then we do some post processing. Lets move everything inside try, and make except block the last thing in function.

Furthermore, `rsp.status_code == 200` isn't resilient enough imho. For example. `rsp.status_code == 204` is also successful. I'd refactor this to look like:

`if rsp.status_code == 404 ....`
`else ... success`

hence we have our happy path always in the line of sight.

One more thing, please check, as I believe `requests` library contains enums for http verbs GET, POST etc.
If this isn't the case, we should introduce our enums - lowers probability of making a typo substantially.